api/src/importers/cors/base-record-utils.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const defaultLog = require('../../utils/logger')('epic-base-record-utils');
+const RecordController = require('./../../controllers/record-controller');
+
+/**
+ * Cors csv base record type handler that can be used directly, or extended if customizations are needed.
+ *
+ * @class BaseRecordUtils
+ */
+class BaseRecordUtils {
+  /**
+   * Creates an instance of BaseRecordUtils.
+   *
+   * @param {*} auth_payload user information for auditing
+   * @param {*} recordType an item from record-type-enum.js -> RECORD_TYPE
+   * @param {*} csvRow an array containing the values from a single csv row.
+   * @memberof BaseRecordUtils
+   */
+  constructor(auth_payload, recordType, csvRow) {
+    if (!recordType) {
+      throw Error('BaseRecordUtils - required recordType must be non-null.');
+    }
+
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+    this.csvRow = csvRow;
+  }
+
+  /**
+   * Transform an Epic record into a NRPTI record.
+   *
+   * Note: Only transforms common fields found in ALL supported epic import types.
+   *       To include other values, extend this class and adjust the object returned by this function as needed.
+   *
+   * @param {object} epicRecord Epic record (required)
+   * @returns {object} NRPTI record.
+   * @throws {Error} if record is not provided.
+   * @memberof BaseRecordUtils
+   */
+  async transformRecord(csvRow) {
+    if (!csvRow) {
+      throw Error('transformRecord - required epicRecord must be non-null.');
+    }
+
+    return {
+      _schemaName: this.recordType._schemaName,
+      recordType: this.recordType.displayName,
+
+      dateAdded: new Date(),
+      dateUpdated: new Date(),
+
+      addedBy: (this.auth_payload && this.auth_payload.preferred_username) || '',
+      updatedBy: (this.auth_payload && this.auth_payload.preferred_username) || '',
+
+      sourceSystemRef: 'cors-csv'
+    };
+  }
+
+  /**
+   * Searches for an existing master record, and returns it if found.
+   *
+   * @param {*} nrptiRecord
+   * @returns {object} existing NRPTI master record, or null if none found or _sourceRefCorsId is null
+   * @memberof BaseRecordUtils
+   */
+  async findExistingRecord(nrptiRecord) {
+    if (!nrptiRecord._sourceRefCorsId) {
+      return null;
+    }
+
+    const masterRecordModel = mongoose.model(this.recordType._schemaName);
+
+    return await masterRecordModel
+      .findOne({
+        _schemaName: this.recordType._schemaName,
+        _sourceRefCorsId: nrptiRecord._sourceRefCorsId
+      })
+      .populate('_flavourRecords');
```

Comment:
Populate only the fields in which you need, ie: `.populate('_flavourRecords', '_id _schemaName')` to reduce memory consumption.